packages/core/src/utils/ComponentError.ts

```diff
@@ -0,0 +1,58 @@
+import {NativeModules} from "react-native";
+
+interface ComponentError extends Error {
+  componentStack: string;
+}
+
+const isJSXError = (error: Error): error is ComponentError => {
+  return "componentStack" in error;
+};
+
+/**
+ * Logs errors with additional React component stack information if available.
+ *
+ * This function checks if the provided error is a React Native rendering error
+ * (i.e., it includes a `componentStack` property). If the `componentStack` is
+ * present and non-empty, the error details, including the message and the component
+ * stack trace, are logged using `EmbraceManager.logHandledError`.
+ *
+ * **Important Notes:**
+ * - The `componentStack` provides a trace of the React component tree at the time
+ *   of the error but does not include line or column numbers.
+ * - This function is complementary to other logging mechanisms that capture more
+ *   detailed stack traces, including line and column numbers.
+ * - By default, in release builds, the screen names in the `componentStack` appear as
+ *   `Unknown`. To retain meaningful screen names in release builds, you must configure
+ *   Metro to preserve the names of React components during minification. For detailed
+ *   instructions, refer to https://docs.embrace.io/docs/react-native/features/tracking-error-component/.
+ *
+ * Example `componentStack` trace:
+ * ```
+ * in App
+ * in RCTView
+ * in Unknown
+ * in AppContainer
+ * ```
+ *
+ * @param error - The error object to be checked and logged.
+ * @returns A promise that resolves to `true` if the error was logged, or `false`
+ * if the error was not a React Native rendering error or the `componentStack` was empty.
+ */
+const logIfComponentError = (error: Error): Promise<boolean> => {
+  if (!isJSXError(error) || error.componentStack === "") {
+    return Promise.resolve(false);
+  }
+  const {message, componentStack} = error;
+
+  const componentStackShrinked = componentStack
+    .replace(/at UNdefined|\(|\)/g, "")
```

Comment:
should define the regex as a constant outside so that it doesn't need to be reevaluated on each call

for the regex itself what is the function of the `|` characters? You could try this version: `.replace(/at undefined \((.*)\)/g, "$1")`

Reply:
This
`/at undefined \((.*)\)/g` delete the whole text, I receive ""